pulp_container/app/access_policy.py

```diff
@@ -51,14 +54,25 @@ def is_private(self, request, view, action):
         return view.get_object().private
 
 
-class NamespaceAccessPolicyFromDB(AccessPolicyFromDB, NamespaceAccessPolicyMixin):
+class NamespacedAccessPolicyFromDB(AccessPolicyFromDB, NamespacedAccessPolicyMixin):
     """
     Access policy for ContainerDistributionViewSet and ContainerPushRepositoryViewSet which handles
     namespace permissions.
     """
 
+class NamespaceAccessPolicy(AccessPolicyFromDB):
+    """
+    Access policy for ContainerNamespaceViewSet.
+    """
 
-class RegistryAccessPolicy(AccessPolicy, NamespaceAccessPolicyMixin):
+    def namespace_is_username(self, request, view, action):
```

Comment:
have not fully loaded this into my head, buy can't we have this part of the `NamespaceAccessPolicyMixin` and NamespaceViewset would just use the `NamespaceAccessPolicyFromDB`?

Reply:
Good question. It's a helper function that only works without exploding when used in an access policy for Namespaces.
And atm NamespaceAccessPolicyFromDB is used only for stuff that is not a namespace. And most of it's helper functions will blow up on namespaces.
That is why i think it should be called "Namespace__d__AccessPolicy".

Having said that, i fail to see, why i introduced:
https://github.com/pulp/pulp_container/pull/244/files#diff-bada7270b35add8957655a936b1bdb701c4dc829033fd87467735943b839c354R29-R31

Reply:
So i have been thinking some and I agree with you, this would have saved some lines of code and we'd have defined `'namespace_is_username` in one place however then  the Mixin would contain all the stuff for repos, distributions and namespaces and this seems to be too entangled. Better to have a separate cleaner policy purely for Namespaces as you did.